bindings/runner/src/common.rs

```diff
@@ -409,6 +409,83 @@ where
     Ok(())
 }
 
+/// Tries to wait for a full arc to show up in the agent infos.
+///
+/// Example:
+/// ```rust
+/// use std::path::Path;
+/// use std::time::Duration;
+/// use holochain_wind_tunnel_runner::prelude::{HolochainAgentContext, HolochainRunnerContext, AgentContext, HookResult, install_app, try_wait_for_min_agents};
+///
+/// fn agent_setup(ctx: &mut AgentContext<HolochainRunnerContext, HolochainAgentContext>) -> HookResult {
+///     install_app(ctx, Path::new("path/to/your/happ").to_path_buf(), &"your_role_name".to_string())?;
+///     if ctx.assigned_behaviour() == "zero" {
+///        try_wait_for_full_arc(ctx)?;
+///     }
+///     Ok(())
+/// }
+/// ```
+pub fn try_wait_for_full_arc<SV>(
+    ctx: &mut AgentContext<HolochainRunnerContext, HolochainAgentContext<SV>>,
+) -> HookResult
+where
+    SV: UserValuesConstraint,
+{
+    let reporter = ctx.runner_context().reporter();
+    let start_discovery = Instant::now();
+
+    loop {
+        // If the wait time specified in the wait_for argument exceeds the
+        // duration that the scenario is supposed to run, we should break
+        // the loop here.
+        if ctx.shutdown_listener().should_shutdown() {
```

Comment:
I'm not sure this is really being invoked. When I tested it locally with a print statement here I didn't see it getting printed out when the scenario got shut down due to exceeding duration before a full arc node was discovered...
But it probably doesn't hurt to have this logic here anyway.